cmd/iam-etcd-store.go

```diff
@@ -313,11 +318,30 @@ func (ies *IAMEtcdStore) loadUser(user string, isSTS bool, m map[string]auth.Cre
 	if u.Credentials.IsExpired() {
 		// Delete expired identity.
 		ctx := ies.getContext()
-		deleteKeyEtcd(ctx, ies.client, getUserIdentityPath(user, isSTS))
-		deleteKeyEtcd(ctx, ies.client, getMappedPolicyPath(user, isSTS, false))
+		deleteKeyEtcd(ctx, ies.client, getUserIdentityPath(user, userType))
+		deleteKeyEtcd(ctx, ies.client, getMappedPolicyPath(user, userType, false))
 		return nil
 	}
 
+	// Check if the rotation of the session key is needed
```

Comment:
This should be done for STS creds which do not have expiry @vadmeste , because it's pointless to rotate an expiring token which will be invalid anyway, a client can request a new one.

Reply:
> This should be done for STS creds which do not have expiry @vadmeste , because it's pointless to rotate an expiring token which will be invalid anyway, a client can request a new one.

oh are you saying there is some STS creds without expiry ? what would be the expiry value in this case ?

Reply:
> oh are you saying there is some STS creds without expiry ? what would be the expiry value in this case ?

because in the old code, expiry seems to be always needed to be specified as claim

Reply:
No if you are doing this just for STS creds it's not needed. Because if you rotate it is not worth it, client won't know know the new session token there is no way to know what was previously generated in STS API.

If the admin creds change then we should simply let these tokens fail.

Reply:
So, I do the rotation only for service accounts now.